src/Module.ts

```diff
@@ -298,6 +300,30 @@ export default class Module {
 			get dynamicImporters() {
 				return dynamicImporters.sort();
 			},
+			get exportedBindings() {
+				const reexportBindings: Record<string, string[]> = {};
+				for (const [name, { source }] of reexportDescriptions) {
+					reexportBindings[source] ??= [];
+					reexportBindings[source].push(name);
+				}
+
+				const exportAllSourceBindings: Record<string, ['*']> = {};
+				for (const source of exportAllSources) {
+					exportAllSourceBindings[source] = ['*'];
+				}
+
+				const exportBindings: Record<string, string[]> =
+					exports.size > 0 ? { '.': [...exports.keys()] } : {};
+
+				return { ...exportBindings, ...reexportBindings, ...exportAllSourceBindings };
```

Comment:
If you do it this way, in a module where you have both an export-all and a regular reexport from the same module

```js
export * from 'foo';
export { foo } from 'foo';
```

as I see it, the export-all would overwrite the reexport in `exportedBindings`, leaving only

```
{ foo: ['*'] }
```

due to the use of spread operators.

But why spread at all, why don't you just create an `exportedBindings` object and put all properties on that one? Then you could just rewrite the second loop like

```js
for (const source of exportAllSources) {
  exportedBindings[source] ??= [];
  exportedBindings[source].push('*');
}
```

For a minimal performance gain (avoiding one computed property access), one could also combine the two lines into one

```js
for (const source of exportAllSources) {
  (exportedBindings[source] ??= []).push('*');
}
```

but only if you think this is still readable.

Reply:
As for the third, I think it would be ok to have an empty array if there are no direct exports, then you just add a property there, simplifying the logic.

Reply:
Ok